src/main/java/com/bookshop/service/impl/BookServiceImpl.java

```
@@ -38,4 +38,17 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(book);
 
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        findById(id);

```

we can throw an exception here in case we can't find book by this id